MIAUI/MIAUI/TasksRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace MIAUI;
+
+public class TasksRepository
+{
+    string _dbPath;
+    public string StatusMessage { get; set; }
+
+    private SQLiteConnection conn;
+    
+    private void Init ()
+    {
+        if (conn != null)
+            return;
+
+        conn = new SQLiteConnection (_dbPath);
+        conn.CreateTable<Model.Task> ();
+    }
+    
+    public TasksRepository (string dbPath)
+    {
+        _dbPath = dbPath;
+    }
+    public async void AddNewTask (string taskName)
+    {
+        int result = 0;
+        try
+        {
+            Init ();
+            // basic validation to ensure a name was entered
+            if (string.IsNullOrEmpty (taskName))
+                throw new Exception ("Valid name required");
+
+            result = conn.Insert (new Model.Task(taskName));
+            result = 0;
+
+            Console.WriteLine (string.Format ("{0} record(s) added (Name: {1})", result, taskName));
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(string.Format("Failed to add {0}. Error: {1}", taskName, ex.Message));
+        }
+    }
+
+    public int UpdateDetails (Model.Task task)
+    {
+        int result = 0;
+        result = conn.Update (task);
+        return result;
```

Comment:
```suggestion
       return conn.Update (task);
```
I think this should work and is less code